system/lib/pthread/library_pthread.c

```diff
@@ -839,6 +839,10 @@ weak_alias(dummy_tsd, __pthread_tsd_main);
 // See system/lib/README.md for static constructor ordering.
 __attribute__((constructor(48)))
 void __emscripten_init_main_thread(void) {
+#ifndef NDEBUG
```

Comment:
Does this mean that `PTHREADS_PROFILING` is now always turned on when building with `ASSERTIONS` enabled? If so, perhaps we should adapt this check?:
https://github.com/emscripten-core/emscripten/blob/7c873832e933e86855f5ef5f7c6438f0e457c94e/emcc.py#L1451-L1454

Reply:
Yes, I had forgotten that this change also enables the thread profiling code in more cases.  i.e. in all builds that use the debug version of libc.

I will update the description, and make sure we decide that is the right way to go.

Reply:
I've updated it so that the code is included in debug builds but only conditionally activated when `PTHREADS_PROFILING` is enabled.